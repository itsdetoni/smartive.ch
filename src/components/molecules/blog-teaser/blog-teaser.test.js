import React from 'react';
import renderer from 'react-test-renderer';

import { BlogTeaser } from './blog-teaser';

describe('Blog Teaser', () => {
  it('renders default correctly', () => {
    const tree = renderer
      .create(<BlogTeaser url="http://example.com" title="Title" subline="Subline" lead="Longer lead text" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders default correctly with an img', () => {
    const tree = renderer
      .create(
        <BlogTeaser url="http://example.com" title="Title" subline="Subline" lead="Longer lead text" img="image.jpg" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
