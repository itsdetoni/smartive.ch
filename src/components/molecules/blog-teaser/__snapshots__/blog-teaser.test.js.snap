// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Teaser renders default correctly 1`] = `
<div
  className="blog-teaser col-xs-12 col-lg-5 first-lg"
  style={Object {}}
>
  <a
    className="blog-teaser__link"
    href="http://example.com"
  >
    <div
      className="blog-teaser__content"
    >
      <h2>
        <small>
          Subline
        </small>
        Title
      </h2>
      <p>
        Longer lead text
      </p>
    </div>
  </a>
</div>
`;

exports[`Blog Teaser renders default correctly with an img 1`] = `
<div
  className="blog-teaser col-xs-12 blog-teaser--image col-lg-7"
  style={
    Object {
      "backgroundImage": "url(image.jpg)",
    }
  }
>
  <a
    className="blog-teaser__link"
    href="http://example.com"
  >
    <div
      className="blog-teaser__content"
    >
      <h2>
        <small>
          Subline
        </small>
        Title
      </h2>
      <p>
        Longer lead text
      </p>
    </div>
  </a>
</div>
`;
